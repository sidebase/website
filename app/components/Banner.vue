<template>
  <div class="w-full py-4 shadow bg-gray-800">
    <div class="container mx-auto flex items-center justify-between">
      <div class="flex items-center gap-3 text-gray-300">
        <i class="pi pi-bolt" style="font-size: 2rem; margin-top: 2px" />
        <div>
          <h1 class="text-lg font-bold">
            We just released our package <span class="font-mono bg-gray-700 p-1 rounded-lg">nuxt-sidebase-parse</span>
          </h1>
          <p class="text-xs">
            A DX focused Nuxt3 package that simplifies data validation and parsing!
          </p>
        </div>
      </div>
      <a href="https://nuxt-sidebase-parse.sidebase.io/index.html" target="_blank">
        <Button label="View documentation" class="p-button-sm" icon="pi pi-book" icon-pos="left" />
      </a>
    </div>
  </div>
</template>

