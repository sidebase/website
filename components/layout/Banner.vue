<template>
  <div class="bg-pickled-bluewood-700 hidden items-center justify-center py-3 text-[1rem] lg:flex">
    <span class="bg-sidebase-green-800 mr-3 rounded-lg px-4 py-1.5 text-xs text-white">New!</span>
    <span class="text-gray-200">
      We just opened the official sidebase Discord server!
    </span>
    <a href="https://discord.gg/VzABbVsqAc" target="_blank" class="text-sidebase-green ml-2 underline">
      Join now!
    </a>
  </div>
</template>
