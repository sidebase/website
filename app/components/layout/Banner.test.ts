import { describe, expect, test } from 'vitest'
import Banner from './Banner.vue'
import { render } from '~/tests/testingLibraryVue'

describe('Banner', () => {
  test('mounts and is still the same', () => {
    const { html } = render(Banner)
    expect(html()).toMatchSnapshot()
  })
})
