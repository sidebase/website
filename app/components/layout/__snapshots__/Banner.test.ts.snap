// Vitest Snapshot v1

exports[`Banner > mounts and is still the same 1`] = `
"<div class=\\"hidden lg:block w-full py-4 shadow bg-gray-800\\">
  <div class=\\"container mx-auto flex items-center justify-between\\">
    <div class=\\"flex items-center gap-3 text-gray-300\\"><i class=\\"pi pi-bolt\\" style=\\"font-size:2rem;margin-top:2px;\\"></i>
      <div>
        <h1 class=\\"text-lg font-bold\\"> We just released our package <span class=\\"font-mono bg-gray-700 p-1 rounded-lg\\">nuxt-sidebase-parse</span></h1>
        <p class=\\"text-xs\\"> A DX focused Nuxt3 package that simplifies data validation and parsing! </p>
      </div>
    </div><a href=\\"https://nuxt-sidebase-parse.sidebase.io/index.html\\" target=\\"_blank\\"><button class=\\"p-button p-component p-button-sm\\" type=\\"button\\">
        <!----><span class=\\"pi pi-book p-button-icon p-button-icon-left\\"></span><span class=\\"p-button-label\\">View documentation</span>
        <!---->
      </button></a>
  </div>
</div>"
`;
